### DISCUSSION TITLE
large "Application Logs"

### DISCUSSION THREAD

**Discussion by andreas-arkulpa** [coollabsio/coolify#332] [Q&A]
Hi @andrasbacsai,

first of all thanks for the amazing tool! I really like coolify.

We have an application which produces a large amount of application logs.
When we want to take a look at the application logs in the coolify dashboard the application crashes. If there is no parameter "since" added to the query it will load all logs sine timestamp 0.

You can see the params here: https://docs.docker.com/engine/api/v1.37/#operation/ContainerLogs

https://github.com/coollabsio/coolify/blob/2971e142698b6f871065ca7ccd02be032adc0bde/src/routes/applications/%5Bid%5D/logs/index.json.ts#L25

Can you think about adding pagination for loading the logs? Maybe it is also an idea to add the "tail" parameter to only load the last x log entries?

You can also tell me where to add the pagination so I can help you with that.

**Comment by andrasbacsai** [comment]
Pagination would be a nice addition to the logs! Thanks for the suggestion!

**Reply by andrasbacsai** [reply]
Converted this to an issue #356, which will be solved in the next version.

**Reply by andreas-arkulpa** [reply]
Thank you very much!!!

**Reply by andrasbacsai** [reply]
The next version was released. Let me know if it's better now. 

There is plenty of room for improvements, but it should be better.

**Reply by andreas-arkulpa** [reply]
Thanks for your help!
It seems that this issue is not fixed, as loading all logs in the docker-host causes the application to crash (there are many large logs). I think it is also needed to add a limit in this line - the param `since` or `tail` should be added here also, if I get it right.
 https://github.com/coollabsio/coolify/blob/2971e142698b6f871065ca7ccd02be032adc0bde/src/routes/applications/%5Bid%5D/logs/index.json.ts#L25

**Reply by andrasbacsai** [reply]
Ah, okay! :) For now, I will limit it to 5000. Is it okay for you? 

I'm currently investigating a possible solution for logging applications into a time-series database (fluentbit + influxdb). After this feature, the logs won't be queried from the containers, but from a database (in real-time).

**Reply by andreas-arkulpa** [reply]
Thanks again for your help!
5000 sounds good - I think it should work.
